src/Traits/DetectsChanges.php

```diff
@@ -38,7 +38,7 @@ public function attributeValuesToBeLogged(string $processingEvent): array
             return [];
         }
 
-        $properties['attributes'] = static::logChanges($this);
+        $properties['attributes'] = static::logChanges($this->fresh() ?? $this);
```

Comment:
I'm assuming `fresh` returns `null` for model instances that haven't been saved yet. Would rather see this as `$this->exists ? $this->fresh() : $this` as that makes it clearer on what is going on.